.github/workflows/build_part_of_image_all_platforms.yaml

```diff
@@ -0,0 +1,84 @@
+# Reusable Workflow for building the Docker Image.
+name: Build Part of Image (All Platforms)
+
+# Controls when the action will run.
+on:
+  # Triggers the workflow when it's referenced in another workflow.
+  workflow_call:
+    # Specify the inputs that need to be passed when calling this workflow.
+    inputs:
+      target:
+        required: true
+        type: string
+      cache_from:
+        required: true
+        type: string
+    secrets:
+      docker_hub_username:
+        required: true
+      docker_hub_password:
+        required: true
+
+jobs:
+  # Build part of the docker image for all platforms.
+  build_all_platforms:
+    name: "Build docker image"
+
+    # The type of runner that the job will run on.
+    runs-on: ubuntu-20.04
+
+    strategy:
+      matrix:
+        platform: ["generic", "nuc8i7beh", "nuc7i7bnh"]
+
+    # Steps represent a sequence of tasks that will be executed as part of the job
+    steps:
+      # Cancels previously running jobs for this workflow
+      - name: Cancel Workflow Action
+        uses: styfle/cancel-workflow-action@0.9.1
```

Comment:
Github now natively supports [concurrency](https://docs.github.com/en/actions/using-jobs/using-concurrency#example-using-concurrency-to-cancel-any-in-progress-job-or-run), so i'd like to move all the workflows over to that as we get to them.

Reply:
I've migrated the workflows over to use the native concurrency feature.